text/0039-service-overhaul.md

```diff
@@ -0,0 +1,196 @@
+
+*Since using the service API was not possible with the actual OBS, any breaking change in it (if there are) will not impact any plugin for OBS.*
+
+# Design
+
+## UI
+The Stream settings page will only contain a combo box listing registered services with a remade "Show All" option. So property view can be shown under it.
+The "short list" with "Show All" option, will be harcoded with a list of service id in the page code.
+Any removed element will be reimplemented in a new way.
+
+(Same for auto-wizard)
+
+## Service plugins
+### Issue
+#### `rtmp-services`
+This plugin provide services with only one protocol, so it need to "register" service like YouTube multiple times (HLS & RTMPS).
+
+And also OBS is too heavily dependent of it.
+
+#### Service integrations
+Those integrations are added and use `rtmp-services` in a hacky way and not as plugins.
+
+### Idea
+#### `rtmp-services`
+This plugin will be put in a state of deprecation with a legacy behavior to not break immediately every setup. And be replaced and finally removed after one major release cycle.
+
+Since OBS is heavily dependent of this plugin, many changes will be needed to remove this dependency.
+
+#### Service integrations
+
+Twitch, YouTube and Restream have their own integration in OBS with OAuth. But those need to be isolated as front-end plugin(s).
+
+With an updated front-end API for those and future third-party plugins.
+
+#### `custom-service`
+This plugin is meant to provide a replacement for `rtmp_custom` type.
+
+If OBS need to be heavily dependent to one plugin, it shall be this one.
+
+The protocol will be detected (or maybe forced set by the user).
+Maybe properties will show depending of the protocol.
+
+#### `obs-services`
+This plugin is meant to provide a replacements for `rtmp_common` type for services who doesn't need custom behavior.
+
+This plugin will need a brand new services.json with new format, to register each streaming service with their own id. No more things like `rtmp-common` id.
+ 
+Those services will be able to provide multiple protocols so no more "Service - HLS" and "Service - FTL".
+
+If a certain protocol is not available, the plugin will not register the service if the service doesn't use another protocol.
+If it does, the missing protocol will not be shown. [Thanks to RFC 38](0038-protocol-api.md).
+
+The notion of maximum and recommended shall be distinguished.
+Maximum, supported resolutions, recommended settings will be ignorable with their respective checkboxes.
+
+Those services should have no specific behavior like ingest management.
+
+But can list with which codec they are compatible if needed. [Thanks to RFC 38](0038-protocol-api.md)
+
+In the end, adding streaming service will only be a JSON addition for this plugin.
+Only improvements will be accepted in the code of this plugin.
+
+#### `obs-services2`
+This plugin is meant to provide a replacements for `rtmp_common` type for services who need a custom behavior.
```

Comment:
If these share code like making a request to a url to get a server list we should add a way to do that from the json. 

Otherwise each of these services should each be split out into their own plugin.